### ISSUE TITLE
Adding different properties to Nodes

### ISSUE THREAD

**Issue by thepunitsingh** [c0fec0de/anytree#88]
Please extend this library so that we can _add different custom properties to the nodes in the tree in python dictionary format and can be able to fetch/operate on them_.
for example, in your tree given in README.rst file, If we would have added the nodes in this way:

```
from anytree import Node, RenderTree
udo = Node("Udo", attributes={age=45, sex='F', sibling_count=3, credit_score=700})
marc = Node("Marc", parent=udo)
lian = Node("Lian", parent=marc, attributes={husband='Garry'})
dan = Node("Dan", parent=udo, attributes={salary='$2500', age=25, sex='M'})
jet = Node("Jet", parent=dan)
jan = Node("Jan", parent=dan)
joe = Node("Joe", parent=dan)
```

The tree generated will be as:
```
Udo {age=45, sex='M', sibling_count=3, credit_score=700}
├── Marc
│   └── Lian {husband='Garry'}
└── Dan {salary='$2500', age=25, sex='M'}
    ├── Jet
    ├── Jan
    └── Joe

```
Further, we can set/get/clear different attributes from the dictionary later as needed as we can do in any python dictionary.

**To summarize**, we can attach different python dictionary to each node and can perform dictionary operations on the attached dictionary.

**Comment by jkbgbr** [comment]
I guess you should try the NodeMixin class.
